sdk/python/touca/_transport.py

```diff
@@ -11,27 +11,11 @@
 
 class Transport:
     def __init__(self):
-        self._token = None
         self._pool = PoolManager(cert_reqs="CERT_REQUIRED", ca_certs=certifi.where())
 
-    def authenticate(self, api_url: str, api_key: str):
-        from touca._options import ToucaError
-
-        if self._token and self._api_key == api_key and self._api_url == api_url:
-            return
-        self._api_key = api_key
+    def configure(self, api_url: str, api_key: str):
```

Comment:
Not sure if configure is a good name for this function. We already use have a `configure` function for the client. I cannot think of a better name though. Can we keep it as authenticate for now?

Reply:
My proposal was based on [this other comment](https://github.com/trytouca/trytouca/pull/491#discussion_r1066892755) to make this function test the given credentials like before. But now that I think more about it, I'd like to slightly modify my suggestion:

Let's keep this function implementation as is. In `update_core_options`, we'd call this function and then add separate logic to test the credentials. In `update_runner_options` we don't need to do the latter because `apply_remote_options` will automatically serve as a way to test the credentials.

Reply:
Update: we had a sync meeting about this and agreed to create a separate endpoint for validating the API Key. The new endpoint would expect the API Key as a custom header and return 204 if it is valid and 403 if not. This way, `/client/signin` will not change. In the future, once all SDKs opt to using the new endpoint, we can remove `/client/signin` for good.

Reply:
Proposal: The post request may include a `team` field in its body. If the field exists, the controller should check if the user with the API Key has permissions to submit to the given team.

- If the Key is invalid, the 403 error should read: 'API Key is invalid'.
- If the team is not accessible, the 403 error should read: 'User is not authorized to access team "team"'.

Thoughts?